state/executor.go

```diff
@@ -450,7 +450,7 @@ func (t *Transition) subGasLimitPrice(msg *types.Transaction) error {
 	factor := new(big.Int)
 	if msg.GasFeeCap != nil && msg.GasFeeCap.BitLen() > 0 {
```

Comment:
There is no strict reason why we should go this way rather than checking tx type. The idea was to be more explicit and avoid potential nil pointer panic issue there.